src/TopoMojo.Api/Features/Gamespace/GamespaceMapper.cs

```diff
@@ -40,7 +38,49 @@ public GamespaceProfile()
 
             CreateMap<QuestionSpec, QuestionView>()
                 .ForMember(d => d.Answer, opt => opt.Ignore())
+                .AfterMap((source, destination) =>
+                {
+                    // if they've solved this question, do include the
+                    // answer for display purposes
+                    if (source.IsCorrect && source.IsGraded)
+                    {
+                        destination.Answer = source.Answer;
+                    }
+                })
             ;
+
+            CreateMap<VariantSpec, VariantView>()
+                .ForMember(d => d.TotalSectionCount, opt => opt.MapFrom(s => s.Sections.Count))
+                .AfterMap((source, destination) =>
```

Comment:
I don't really love this kind of thing either, honestly. The goal was to encapsulate the rule of "`VariantViews` may not have all the sections" and make it "impossible" to map one with sections that shouldn't be visible, but I have similar reservations. This actually gets into why I'm constantly fighting an internal battle about automapper 🤣. Anyway, I made it an instance method of `VariantView` since it mutates the object, but if you'd rather an extension, we can do that too.

Reply:
Moved to extension